### ISSUE TITLE
Backup does not work ... clicking on the button does nothing .. popup dialog does not open (Betterbird 115; Ubuntu)

### ISSUE THREAD

**Issue by wwender** [RealRaven2000/quickFilters#245]
Thunderbird: Betterbird 115.12.0-bb29 (64-bit)
OS: Ubuntu 22

Debug Log Messages  (XXX for hiding personal information) : 

---

```
quickFilters [logTime init]
 Adding FilterList... [qFilters-utils.js:720:13](chrome://quickfilters/content/qFilters-utils.js)
Manually calling notifyTools.enable() is no longer needed. [notifyTools.js:145:13](jar:file:///home/XXXXX/.thunderbird/XXXXX.XXXXX/extensions/quickFilters@axelg.com.xpi!/chrome/content/scripts/notifyTools.js)
quickFilters  quickFilters.Init()  
        Assistant active: false
        BrowserInfo: [object Object]
```

---

There is a list of messages like :  # 1. Added filter to JSON: XXXX

---

```
quickFilters 11:6:43.457  [0 ms]   
 24. Added filter to JSON: XXXXX
quickFilters 11:6:43.458  [1 ms]   
 popupAlert(Archiving 24 Filters, quickFilters) [qFilters-utils.js:720:13](chrome://quickfilters/content/qFilters-utils.js)
Uncaught (in promise) NS_ERROR_FILE_UNRECOGNIZED_PATH: Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIFile.initWithPath]
    fileFilters chrome://quickfilters/content/qFilters-list.js:1710
    store chrome://quickfilters/content/qFilters-list.js:1902
    onclick chrome://messenger/content/FilterListDialog.xhtml:1
```

---

   No popup is shown ... just a silent dead ...

**Comment by wwender** [comment]
I think I solved it : 

```
let dPath = prefs.getStringPref('files.path');
		if (dPath) {
			let defaultPath = Cc["@mozilla.org/file/local;1"].createInstance(NSIFILE);
			defaultPath.initWithPath(dPath);

```
There is a config setting "extensions.quickfilters.files.path"  this seems to have a value from a former windows system (the profile had also been used on a windows system) ... after I set extensions.quickfilters.files.path it worked .

**Comment by RealRaven2000** [comment]
thanks for reporting this, I can make this more robust for the future.

**Comment by wwender** [comment]
👍

**Comment by RealRaven2000** [comment]
I will look at a workaround anyway it should not fail silently even if the path is garbage. Will close the issue myself once it is implemented and a new version released that contains the fix..

**Comment by llavegrifa** [comment]
Opensuse tumbleweed installation. 
Back up opens a small  message panel saying so many filters saved and the closes. I cant find where/what is saved as no file name or directory is requested.
Restore button won't do anything.

**Comment by RealRaven2000** [comment]
> Opensuse tumbleweed installation. Back up opens a small message panel saying so many filters saved and the closes. I cant find where/what is saved as no file name or directory is requested. Restore button won't do anything.

can you try the latest version below, and then also switch on debug mode (quickFilters settings >> Advanced tab). Use the wheel beside _Debug Mode_ and also toggle the value for `debug.filterList`

<img width="787" height="554" alt="Image" src="https://github.com/user-attachments/assets/a9b097d5-5d55-4aad-a307-171a1737264a" />

You can then capture a console log with CTRL +SHIFT + J and right-click Copy All messages when testing the [Backup Filters] button

<img width="745" height="396" alt="Image" src="https://github.com/user-attachments/assets/63c11502-e5d7-42b7-8c40-06ff3a458895" />

latest version:
[quickFilters-6.13.2pre18.zip](https://github.com/user-attachments/files/30417471/quickFilters-6.13.2pre18.zip)

----

To install version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that).

**Comment by llavegrifa** [comment]
```log
Access attempt to RS collection "main/moz-essential-domain-fallbacks" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/ai-window-prompts" from local dump was blocked/filtered. 3 Utils.sys.mjs:304:13
Access attempt to RS collection "main/fxmonitor-breaches" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/search-config-overrides-v2" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/search-telemetry-v2" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/fxmonitor-breaches" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/newtab-wallpapers-v2" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/fxmonitor-breaches" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/urlbar-persisted-search-terms" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/fxmonitor-breaches" from local dump was blocked/filtered. 2 Utils.sys.mjs:304:13
BackupService: There was an error while trying to get the Document's directory NS_ERROR_FAILURE: Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIProperties.get]
    get docsDirFolderPath resource:///modules/backup/BackupService.sys.mjs:1299
    get DEFAULT_PARENT_DIR_PATH resource:///modules/backup/BackupService.sys.mjs:1027
    takeMeasurements resource:///modules/backup/BackupService.sys.mjs:4381
BackupService.sys.mjs:1301:23
NotAllowedError: PathUtils.join: PathUtils does not support empty paths BackupService.sys.mjs:4380:17
Access attempt to RS collection "main/newtab-frecency-boosted-sponsors" from local dump was blocked/filtered. 3 Utils.sys.mjs:304:13
Access attempt to RS collection "main/fxmonitor-breaches" from local dump was blocked/filtered. 3 Utils.sys.mjs:304:13
The resource at “https://private-user-images.githubusercontent.com/favicon.ico” was blocked by OpaqueResponseBlocking. Reason: “after sniff: nosniff is true”.
Access attempt to RS collection "main/fxmonitor-breaches" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Access attempt to RS collection "main/newtab-frecency-boosted-sponsors" from local dump was blocked/filtered. Utils.sys.mjs:304:13
Key event not available on some keyboard layouts: key=“z” modifiers=“control,alt” id=“toggleSidebarKb” browser.xhtml
Key event not available on some keyboard layouts: key=“u” modifiers=“accel,alt” id=“viewOpenTabsSidebarKb” browser.xhtml
Key event not available on some keyboard layouts: key=“x” modifiers=“accel,alt” id=“viewGenaiChatSidebarKb” browser.xhtml
Key event not available on some keyboard layouts: key=“r” modifiers=“accel,alt” id=“key_toggleReaderMode” browser.xhtml
Key event not available on some keyboard layouts: key=“i” modifiers=“accel,alt,shift” id=“key_browserToolbox” browser.xhtml
   
```

**Comment by RealRaven2000** [comment]
@llavegrifa 
thanks for the log, try the new version below. It should either work or give us additional info...
Please enable debug mode in quickFilters settings >> Advanced:

<img width="876" height="346" alt="Image" src="https://github.com/user-attachments/assets/9fea603b-b31d-4055-a1c3-67b5c0da8b7d" />

[quickFilters-6.13.2pre30.zip](https://github.com/user-attachments/files/30461747/quickFilters-6.13.2pre30.zip)

---

To install version above download zip file and drag the file into Thunderbird Add-ons Manager (do not extract contents, it won't install like that).

**Comment by llavegrifa** [comment]
Excellent: version 6.13.pre30 works ok.

**Comment by RealRaven2000** [comment]
Fixed in 6.13.2 - Published 07/08/2026
